src/solutions/optimization_solutions.jl

```diff
@@ -17,28 +17,38 @@ Representation of the solution to an nonlinear optimization defined by an Optimi
   callback (`sol.retcode === :Terminated`), or whether it exited due to an error. For more
   details, see the return code section of the DifferentialEquations.jl documentation.
 """
-struct OptimizationSolution{T, N, uType, P, A, Tf, O} <: AbstractOptimizationSolution{T, N}
+struct OptimizationSolution{T, N, uType, C <: AbstractOptimizationCache, A, Tf, O} <:
```

Comment:
this will likely cause a test failure?

Reply:
Yes it will.

Reply:
We could make a `build_solution` for `AbstracOptimizationProblem`, wrap the problem it in a `DefaultOptimizationCache` and warn the user that he should implement a cache interface.

Reply:
Also the `AbstractOptimizationCache` needs a big docstring, explaining what we expect from it.

Reply:
Yes, I think we need to be much more careful here. I didn't catch this before, but we should make sure that this has as nice of an upgrade path as possible, because otherwise there will be a version mismatch issue